### PR TITLE
Add feature of Storage Pool Options (1/3)

### PR DESCRIPTION
**Pull request by vatsa287**
**What this PR does / why we need it**:
This PR implements Storage Pool Options support for Kadalu,
This particular PR handles CRD, Server, Operator changes needed to support for adding options which is handled by `handle_added`. Mount-Options, Updation(handle_modified) and CLI changes will be made in follow-up PRs.

Changes:
- In CRDs introduce `options` as array of objects.
- In Operator validate these options and add to configmap on `handle_added`.
- In Server fetch options from `.info` file and pass it to `kadalu-volgen` for creation of storage-unit & client volfiles.

**Which issue(s) this PR fixes**:
Updates #982 

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: server/serverutils.py [hunk 1]
```diff
@@ -108,7 +118,7 @@ def generate_shd_volfile(data, shd_volfile_path):
 def generate_client_volfile(data, client_volfile_path):
     """ Generate client volfile using Kadalu Volgen library"""
 
-    options = copy.copy(DEFAULT_OPTIONS)
+    options = copy.copy(DEFAULT_OPTIONS.update(data["options"]))
```
Comment (aravindavk) [inline server/serverutils.py]: Do not update DEFAULT_OPTIONS directly

```
options = copy.copy(DEFAULT_OPTIONS)
options.update(data["options"])
```

**Comment by aravindavk** [comment]
Please post the sample yaml file used

File: server/serverutils.py (hunk 1 above)
Comment (vatsa287) [reply server/serverutils.py]: Ack

**Comment by vatsa287** [comment]
kadalu-storage.yaml

```
apiVersion: kadalu-operator.storage/v1alpha1
kind: KadaluStorage
metadata:
  name: storage-pool-1
spec:
  type: Replica3
  options:
    - key: "performance.open-behind"
      value: "on"
    - key: "performance.quick-read"
      value: "on"
    - key: "diagnostics.storage-unit-log-level"
      value: "debug"
    - key: "diagnostics.client-log-level"
      value: "debug"
    - key: "performance.write-behind"
      value: "on"
  storage:
    - node: thinkpad      # node name as shown in `kubectl get nodes`
      path: /exports/vol1/s1 # Device to provide storage to all PVs

    - node: thinkpad      # node name as shown in `kubectl get nodes`
      path: /exports/vol1/s2

    - node: thinkpad      # node name as shown in `kubectl get nodes`
      path: /exports/vol1/s3
  ```
  
Diff of client-volfile with and without options https://www.diffchecker.com/9h6gvAKt/

**Review by aravindavk — APPROVED** [review]

**Pull request merged**